modules/core/test/v2/unit/coins/sol.ts

```diff
@@ -72,8 +72,92 @@ describe('SOL:', function () {
   }));
 
   describe('Parse Transactions:', () => {
-    it('should parse a transfer transaction', async function () {
-      await should.throws(() => basecoin.parseTransaction('placeholder'), 'parseTransaction method not implemented');
+    it('should parse an unsigned transfer transaction', async function () {
+      const parsedTransaction = await basecoin.parseTransaction({
+        txBase64: testData.rawTransactions.transfer.unsigned,
+        feeInfo: {
+          fee: '5000',
+        },
+      });
+
+      parsedTransaction.should.deepEqual({
+        inputs: {
+          address: 'CP5Dpaa42RtJmMuKqCQsLwma5Yh3knuvKsYDFX85F41S',
+          amount: 305000,
+        },
+        outputs: [
+          {
+            address: 'CP5Dpaa42RtJmMuKqCQsLwma5Yh3knuvKsYDFX85F41S',
+            amount: '300000',
+          },
+        ],
+      });
+    });
+
+    it('should parse an signed transfer transaction', async function () {
+      const parsedTransaction = await basecoin.parseTransaction({
+        txBase64: testData.rawTransactions.transfer.signed,
+        feeInfo: {
+          fee: '5000',
+        },
+      });
+
+      parsedTransaction.should.deepEqual({
+        inputs: {
+          address: 'CP5Dpaa42RtJmMuKqCQsLwma5Yh3knuvKsYDFX85F41S',
+          amount: 305000,
+        },
+        outputs: [
+          {
+            address: 'CP5Dpaa42RtJmMuKqCQsLwma5Yh3knuvKsYDFX85F41S',
+            amount: '300000',
+          },
+        ],
+      });
+    });
+
+    it('should parse an unsigned transfer transaction', async function () {
+      const parsedTransaction = await basecoin.parseTransaction({
+        txBase64: testData.rawTransactions.walletInit.unsigned,
+        feeInfo: {
+          fee: '5000',
+        },
+      });
+
+      parsedTransaction.should.deepEqual({
+        inputs: {
+          address: '8Y7RM6JfcX4ASSNBkrkrmSbRu431YVi9Y3oLFnzC2dCh',
+          amount: 310000,
```

Comment:
I'm not following why here the fee is charged on the input as 10000 and the feeInfo is 5000 lamports.

Reply:
The input fee is the cost per signature, and since its a walletInit it has 2 signatures

Reply:
Excellent.